From: Nix Packaging
Subject: Use portable TMPDIR instead of hardcoded /var/tmp

Several PCP build scripts hardcode /var/tmp for temporary files. This fails
in the Nix sandbox which restricts filesystem access - /var/tmp is not
writable during Nix builds.

This patch changes /var/tmp to ${TMPDIR:-/tmp} which:
- Uses $TMPDIR when set (Nix sets this to a writable sandbox directory)
- Falls back to /tmp on traditional systems where TMPDIR may not be set
- Is POSIX-compliant and improves portability across build environments

Affected files (build-time scripts):
- src/pmdas/bind2/mk.rewrite
- src/pmdas/jbd2/mk.rewrite
- src/pmdas/linux/mk.rewrite
- src/pmdas/linux_proc/mk.rewrite
- src/bashrc/getargs
- src/libpcp/src/check-errorcodes
- src/libpcp3/src/check-errorcodes
- src/libpcp/doc/mk.cgraph
- src/pmlogcompress/check-optimize

--- a/src/pmdas/bind2/mk.rewrite
+++ b/src/pmdas/bind2/mk.rewrite
@@ -11,7 +11,7 @@
 # and generate pmlogrewrite(1) rules to force the PMIDs to be correct.
 #

-tmp=/var/tmp/mk.rewrite-$$
+tmp=${TMPDIR:-/tmp}/mk.rewrite-$$
 status=0
 trap "rm -f $tmp.*; exit \$status" 0 1 2 3 15

--- a/src/pmdas/jbd2/mk.rewrite
+++ b/src/pmdas/jbd2/mk.rewrite
@@ -6,7 +6,7 @@
 #	platform dependent.
 #

-tmp=/var/tmp/$$
+tmp=${TMPDIR:-/tmp}/$$
 trap "rm -f $tmp.*; exit 0" 0 1 2 3 15

 cat <<End-of-File >$tmp.c
--- a/src/pmdas/linux/mk.rewrite
+++ b/src/pmdas/linux/mk.rewrite
@@ -6,7 +6,7 @@
 #	platform dependent.
 #

-tmp=/var/tmp/$$
+tmp=${TMPDIR:-/tmp}/$$
 trap "rm -f $tmp.*; exit 0" 0 1 2 3 15

 cat <<End-of-File >$tmp.c
--- a/src/pmdas/linux_proc/mk.rewrite
+++ b/src/pmdas/linux_proc/mk.rewrite
@@ -6,7 +6,7 @@
 #	platform dependent.
 #

-tmp=/var/tmp/$$
+tmp=${TMPDIR:-/tmp}/$$
 trap "rm -f $tmp.*; exit 0" 0 1 2 3 15

 cat <<End-of-File >$tmp.c
--- a/src/bashrc/getargs
+++ b/src/bashrc/getargs
@@ -17,7 +17,7 @@ then
 exit 1
 fi

-tmp=/var/tmp/bashrc-getargs-$$
+tmp=${TMPDIR:-/tmp}/bashrc-getargs-$$
 status=1
 trap "rm -f $tmp.*; exit \$status" 0 1 2 3 15

--- a/src/libpcp/src/check-errorcodes
+++ b/src/libpcp/src/check-errorcodes
@@ -6,7 +6,7 @@
 # Check error codes exposure in QA tests.
 #

-tmp=/var/tmp/check-errorcodes-$$
+tmp=${TMPDIR:-/tmp}/check-errorcodes-$$
 rm -f $tmp.err
 trap "sts=0; [ -f $tmp.err ] && sts=1; rm -f $tmp.*; exit \$sts" 0 1 2 3 15

--- a/src/libpcp3/src/check-errorcodes
+++ b/src/libpcp3/src/check-errorcodes
@@ -6,7 +6,7 @@
 # Check error codes exposure in QA tests.
 #

-tmp=/var/tmp/check-errorcodes-$$
+tmp=${TMPDIR:-/tmp}/check-errorcodes-$$
 rm -f $tmp.err
 trap "sts=0; [ -f $tmp.err ] && sts=1; rm -f $tmp.*; exit \$sts" 0 1 2 3 15

--- a/src/libpcp/doc/mk.cgraph
+++ b/src/libpcp/doc/mk.cgraph
@@ -5,7 +5,7 @@
 # Assumes libpcp source and cscope index are in ../src
 #

-tmp=/var/tmp/$$
+tmp=${TMPDIR:-/tmp}/$$
 trap "rm -f $tmp.*; exit \$sts" 0 1 2 3 15

 rm -f $tmp.*
--- a/src/pmlogcompress/check-optimize
+++ b/src/pmlogcompress/check-optimize
@@ -9,7 +9,7 @@
 # below there)
 #

-tmp=/var/tmp/check-optimize-$$
+tmp=${TMPDIR:-/tmp}/check-optimize-$$
 here=`pwd`

 verbose=false

